pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -187,3 +237,140 @@ func getPacketInfo(pktIn *ofctrl.PacketIn, ob *logInfo) error {
 	}
 	return nil
 }
+
+// rejectRequest sends reject response to the requesting client, based on the
+// packet-in message.
+func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
+
+	var (
+		srcMAC     net.HardwareAddr
+		dstMAC     net.HardwareAddr
+		srcIP      string
+		dstIP      string
+		IPProtocol uint8
+		TCPSrcPort uint16
+		TCPDstPort uint16
+		TCPFlags   uint8
+		TCPSeqNum  uint32
+		TCPAckNum  uint32
+		ICMPType   uint8
+		ICMPCode   uint8
+		ICMPData   []byte
+		inPort     uint32
+		outPort    uint32
+	)
+
+	// Get ethernet data.
+	srcMAC = pktIn.Data.HWDst
+	dstMAC = pktIn.Data.HWSrc
+	IPProtocol = uint8(pktIn.Data.Ethertype)
+
+	switch pktIn.Data.Ethertype {
+	case protocol.IPv4_MSG:
+		// Get IP data.
+		ipv4In := pktIn.Data.Data.(*protocol.IPv4)
+		srcIP = ipv4In.NWDst.String()
+		dstIP = ipv4In.NWSrc.String()
+		switch ipv4In.Protocol {
+		case protocol.Type_TCP:
+			IPProtocol = protocol.Type_TCP
+
+			// Transfer Buffer to TCP.
+			tcpBytes, err := ipv4In.Data.(*util.Buffer).MarshalBinary()
+			if err != nil {
+				return err
+			}
+			tcpIn := new(protocol.TCP)
+			err = tcpIn.UnmarshalBinary(tcpBytes)
+			if err != nil {
+				return err
+			}
+
+			// Set TCP Reset packet data.
+			TCPSeqNum = 0
+			TCPAckNum = tcpIn.SeqNum + 1
+			TCPSrcPort = tcpIn.PortDst
+			TCPDstPort = tcpIn.PortSrc
+			TCPFlags = TCPRst + TCPAck
+		default:
```

Comment:
What do we do about UDP traffic?

Reply:
We return ICMP host administratively prohibited, which is also how KVM impl it.